src/cpp/session/modules/SessionDiagnostics.cpp

```diff
@@ -866,12 +867,43 @@ Error lintRSourceDocument(const json::JsonRpcRequest& request,
             documentId,
             isExplicit,
             isFragment);
-   
-   pResponse->setResult(lintAsJson(results.lint()));
-   
+
+   std::vector<module_context::SourceMarker> markers =
+       modules::markers::markersForFile(documentPath);
+   LintItems lintItems = results.lint();
+
+   using namespace module_context;
+   for (SourceMarker marker : markers)
```

Comment:
Can you use `const&` here?

```suggestion
   for (const SourceMarker& marker : markers)
```

Or even, `const auto&` if you're lazy.